File: app/src/main/java/com/qiaoqiao/repository/web/ui/WebLinkFragment.java
package com.qiaoqiao.repository.web.ui;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.qiaoqiao.R;
import com.qiaoqiao.databinding.FragmentWebLinkBinding;

public final class WebLinkFragment extends Fragment implements View.OnClickListener {
	/**
	 * Main layout for this component.
	 */
	private static final int LAYOUT = R.layout.fragment_web_link;
	private FragmentWebLinkBinding mBinding;

	public static WebLinkFragment newInstance(@NonNull Context cxt) {
		return (WebLinkFragment) WebLinkFragment.instantiate(cxt, WebLinkFragment.class.getName());
	}

	@Nullable
	@Override
	public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		mBinding = DataBindingUtil.inflate(inflater, LAYOUT, container, false);
		mBinding.uriTv.setText(R.string.demo_web);
		mBinding.takeUriBtn.setOnClickListener(this);
		return mBinding.getRoot();
	}


	@Override
	public void onClick(View v) {
		Intent data = new Intent();
		data.setData(Uri.parse(mBinding.uriTv.getText()
		                                     .toString()));
		getActivity().setResult(Activity.RESULT_OK, data);
		getActivity().supportFinishAfterTransition();
	}
}

File: app/src/main/java/com/qiaoqiao/repository/web/ui/WebLinkFragment.kt
package com.qiaoqiao.repository.web.ui

import android.app.Activity
import android.content.Intent
import android.databinding.DataBindingUtil
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.qiaoqiao.R
import com.qiaoqiao.databinding.FragmentWebLinkBinding

class WebLinkFragment : Fragment(), View.OnClickListener {
    private val LAYOUT = R.layout.fragment_web_link
    private var binding: FragmentWebLinkBinding? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate(inflater, LAYOUT, container, false)
        binding?.uriTv?.setText(R.string.demo_web)
        binding?.takeUriBtn?.setOnClickListener(this)
        return binding?.root
    }

    override fun onClick(v: View) {
        val data = Intent()
        data.data = Uri.parse(binding?.uriTv?.text.toString())
        with(activity) {
            setResult(Activity.RESULT_OK, data)
            supportFinishAfterTransition()
        }
    }
}